stratum/hal/lib/barefoot/bfrt_table_manager.cc

```diff
@@ -486,10 +486,33 @@ ::util::StatusOr<::p4::v1::TableEntry> BfrtTableManager::BuildP4TableEntry(
     result.mutable_action()->set_action_profile_group_id(selector_group_id);
   }
 
-  // Counter data
-  uint64 bytes, packets;
-  if (table_data->GetCounterData(&bytes, &packets).ok()) {
-    if (request.has_counter_data()) {
+  // Counter data, if applicable.
+  // We check for direct resources, since some calls to this function set the
+  // counter_data field for tables without direct counters.
+  if (request.has_counter_data() && table.direct_resource_ids_size()) {
+    for (const int id : table.direct_resource_ids()) {
+      uint64 bytes = 0, packets = 0;
+      ASSIGN_OR_RETURN(auto direct_counter,
+                       p4_info_manager_->FindDirectCounterByID(id));
```

Comment:
This will fail if there is more than one direct resource on a table. Can this happen? Should we check this here or on pipeline push?

Reply:
direct meter + direct counter?

Reply:
maybe we don't return if the `id` is not a direct counter id, but just skip/continue if it is not for direct counter?